cargo-dist/src/build/cargo.rs

```diff
@@ -66,7 +68,8 @@ impl<'a> DistGraphBuilder<'a> {
         }
 
         let mut builds = vec![];
-        for (target, binaries) in targets {
+        for (target_triple, binaries) in targets {
+            let target = target_triple.parse().unwrap();
```

Comment:
Given we're still in the plan step, and this returns a `DistResult`, we can safely return an error here if we get an unparseable triple. Just need to make sure we return a user-legible error.